MekHQ/src/mekhq/gui/dialog/CustomizePersonDialog.java

```diff
@@ -581,6 +588,28 @@ public Component getListCellRendererComponent(final JList<?> list,
             y++;
         }
 
+        if (campaign.getCampaignOptions().useRetirementDateTracking() && (retirement != null)) {
```

Comment:
Seems like it'd be possible for retirement to be null, but the user wants to explicitly set a retirement date. This logic would disallow that workflow.

Reply:
Same logic is used for time in service and time in rank.